src/sympc/protocol/falcon/falcon.py

```diff
@@ -361,11 +391,51 @@ def multiplication_protocol(
         Returns:
             shares (ReplicatedSharedTensor): results in terms of ReplicatedSharedTensor.
         """
-        op = getattr(operator, op_str)
-
-        z_value = (
-            op(x.shares[0], y.shares[0], **kwargs)
-            + op(x.shares[1], y.shares[0], **kwargs)
-            + op(x.shares[0], y.shares[1], **kwargs)
+        op = ReplicatedSharedTensor.get_op(x.ring_size, op_str)
+
+        z_value = shares_sum(
+            [
+                op(x.shares[0], y.shares[0], **kwargs),
+                op(x.shares[1], y.shares[0], **kwargs),
+                op(x.shares[0], y.shares[1], **kwargs),
+            ],
+            x.ring_size,
         )
         return z_value
+
+    @staticmethod
+    def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
```

Comment:
Q: Can we add a check if "b" has `ring_size` equal with 2 and throw an exception if not.